test/mechanics/timeIntegration/TimeControl.cpp

```diff
@@ -0,0 +1,92 @@
+#include "BoostUnitTest.h"
+
+#include "mechanics/timeIntegration/TimeControl.h"
+
+#include <iostream>
+
+using namespace NuTo;
+
+// %%% Helper functions %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+bool CheckExceptionMessageCorrect(MechanicsException const& ex,std::string expectedMsg)
```

Comment:
I think, you should use the BOOST_CHECK_EQUAL macro instead of this method,
something like `BOOST_CHECK_EQUAL(GetRefinedErrorMsg(ex), expectedMsg);`
Or provide further comments on why you did it with a custom implementation.